src/argilla/client/feedback/dataset/remote/mixins.py

```diff
@@ -0,0 +1,120 @@
+
+        Returns:
+            Either the record of the given index, or a list with the records at the given indexes.
+        """
+        offsets = []
+        limit = None
+        num_records = len(self)
+        if isinstance(key, slice):
+            start, stop, step = key.indices(num_records)
+            if step is not None and step != 1:
+                raise ValueError("When providing a `slice` just `step=None` or `step=1` are allowed.")
+            if start < 0:
+                start += num_records
+            if stop < 0:
+                stop += num_records
+            if start < 0 or stop < 0:
+                raise IndexError(
+                    f"Index {start if start < 0 else stop} is out of range, dataset has {num_records} records."
+                )
+            limit = stop - start
+            if limit < 0:
+                raise ValueError("Negative slice bounds are not supported.")
+            elif limit < FETCHING_BATCH_SIZE:
+                offsets = [start]
+                limits = [limit]
+            else:
+                offsets = list(range(start, stop, FETCHING_BATCH_SIZE))
+                limits = [FETCHING_BATCH_SIZE] * len(offsets)
+                if stop % FETCHING_BATCH_SIZE != 0:
+                    offsets[-1] = stop - (stop % FETCHING_BATCH_SIZE) + 1
+                    limits[-1] = (stop % FETCHING_BATCH_SIZE) - 1
+        elif isinstance(key, int):
+            if key < 0:
+                key += num_records
+            if key < 0 or key >= num_records:
+                raise IndexError(f"Index {key} is out of range, dataset has {num_records} records.")
+            offsets = [key]
+            limits = [1]
+        else:
+            raise TypeError("Only `int` and `slice` are supported as index.")
+
+        records = []
+        for offset, limit in zip(offsets, limits):
+            fetched_records = datasets_api_v1.get_records(
+                client=self._client,
+                id=self._dataset_id,
+                offset=offset,
+                limit=limit,
+                **({} if not hasattr(self, "_filters") else self._filters),
+            ).parsed
+            records.extend([self._parse_record(record) for record in fetched_records.items])
+        return records[0] if isinstance(key, int) else records
+
+    @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
+    def __iter__(
+        self: Union["RemoteFeedbackRecords", "FilteredRemoteFeedbackRecords"]
+    ) -> Iterator[RemoteFeedbackRecord]:
+        """Iterates over the `FeedbackRecord`s of the current `FeedbackDataset` in Argilla."""
+        current_batch = 0
+        while True:
+            batch = datasets_api_v1.get_records(
+                client=self._client,
+                id=self._dataset_id,
+                offset=FETCHING_BATCH_SIZE * current_batch,
+                limit=FETCHING_BATCH_SIZE,
+                **({} if not hasattr(self, "_filters") else self._filters),
+            ).parsed
```

Comment:
You can use the same extracted method here. No need for logic duplication (filters) and the component becomes more extensible.